src/json.rs

```diff
@@ -62,19 +60,22 @@ where
 }
 
 #[cfg(all(feature = "gateway", not(feature = "simd-json")))]
-pub(crate) fn from_str<'a, T>(s: &'a mut str) -> Result<T>
+pub(crate) fn from_str<T>(s: &str) -> Result<T>
 where
-    T: Deserialize<'a>,
+    T: DeserializeOwned,
 {
     Ok(serde_json::from_str(s)?)
 }
 
 #[cfg(all(feature = "gateway", feature = "simd-json"))]
-pub(crate) fn from_str<'a, T>(s: &'a mut str) -> Result<T>
+pub(crate) fn from_str<T>(s: &str) -> Result<T>
 where
-    T: Deserialize<'a>,
+    T: DeserializeOwned,
 {
-    Ok(simd_json::from_str(s)?)
+    // Have to clone the argument because simd_json mutates the argument
+    // And we can't just take &mut str either because simd_json mutates the string such that it may
+    // not be UTF-8 afterwards, which is unsound
+    Ok(simd_json::from_slice(&mut s.as_bytes().to_vec())?)
```

Comment:
Does `from_slice` work with multi-byte characters? I think that passing in a `&mut s.to_string()` would still be fine here, since even if the string becomes invalid UTF-8, it gets dropped immediately afterward.

Reply:
> Does from_slice work with multi-byte characters?

Yes:
![Screenshot_20230919_033925](https://github.com/serenity-rs/serenity/assets/21220820/3a7f498c-e9af-4d73-abf8-a63b8e31aa02)

> I think that passing in a &mut s.to_string() would still be fine here, since even if the string becomes invalid UTF-8, it gets dropped immediately afterward.

I originally thought the mere existence of a non UTF-8 str is instant UB but I looked it up and it seems you're right

Still, that would need an unsafe block. Using `from_slice` needs no unsafe block and has no performance penalty (I expect?